src/Array.mo

```diff
@@ -723,12 +723,124 @@ module {
   ///
   /// let array = [1,2,3,4,5];
   /// let subArray = Array.subArray<Nat>(array, 2, 3);
+  /// ```
   /// Runtime: O(length);
   /// Space: O(length);
-  public func subArray<X>(arr: [X], start: Nat, length: Nat): [X] {
+  public func subArray<X>(arr : [X], start : Nat, length : Nat) : [X] {
     if (start + length > arr.size()) { Prim.trap("Array.subArray") };
-    tabulate<X>(length, func(i) {
-      arr[start + i]
-    });
+    tabulate<X>(
+      length,
+      func(i) {
+        arr[start + i]
+      }
+    )
   };
-};
+
+  /// Returns the index of the first `needle` element in the `haystack` array.
+  ///
+  /// ```motoko include=import
+  /// import Char "mo:base/Char";
+  /// let array = ['c', 'o', 'f', 'f', 'e', 'e'];
+  /// assert Array.indexOf<Char>(array, 'c', Char.equal) == ?0;
+  /// assert Array.indexOf<Char>(array, 'f', Char.equal) == ?2;
+  /// assert Array.indexOf<Char>(array, 'g', Char.equal) == null;
+  /// ```
+  ///
+  /// Runtime: O(haystack.size());
+  /// Space: O(1);
+  public func indexOf<X>(haystack : [X], needle : X, eq : (X, X) -> Bool) : ?Nat = nextIndexOf<X>(haystack, needle, 0, eq);
```

Comment:
Is it possible to rename the arguments to `element`, `array`, and `equal` for unification reasons? 

https://github.com/dfinity/motoko-base/blob/e5eec2c9b4761209a8f1e95abb350d81b362744e/src/Buffer.mo#L1041

I believe some other methods might have equivalents in that module as well.

Reply:
Sure, done.

Reply:
@kentosugama, I wonder whether we should also swap the argument order for consistency:

```
public func indexOf<X>(element : X, buffer : Buffer<X>, equal : (X, X) -> Bool) : ?Nat { 
public func indexOf<X>(array : [X], element : X, equal : (X, X) -> Bool) : ?Nat {
```

Reply:
Done!